src/goto-programs/lazy_goto_model.cpp

```diff
@@ -88,6 +88,25 @@ lazy_goto_modelt::lazy_goto_modelt(lazy_goto_modelt &&other)
 }
 //! @endcond
 
+/// Performs initial symbol table and `language_filest` initialization from
+/// a given commandline and parsed options. This is much the same as the
+/// initial parsing phase of `initialize_goto_model`, except that all function
+/// bodies may be left blank until `get_goto_function` is called for the first
+/// time. This *must* be called before `get_goto_function` is called.
+///
+/// Whether they are in fact left blank depends on the language front-end
+/// responsible for a particular function: it may be fully eager, in which case
+/// only the `goto_convert` phase is performed lazily, or fully lazy, in which
+/// case no function has a body until `get_goto_function` is called, or anything
+/// in between. At the time of writing only Java-specific frontend programs use
+/// `lazy_goto_modelt`, so only `java_bytecode_languaget` is relevant, and that
+/// front-end supplies practically all methods lazily, apart from
+/// `__CPROVER__start` and `__CPROVER_initialize`. In general check whether a
+/// language implements `languaget::convert_lazy_method`; any method not handled
```

Comment:
Can you maybe provide an exact code snippet showing how to do this? This is a bit vague as-is, while at the same time one of the most important bits when it comes to documenting this interface.

Reply:
Added further docs to `get_goto_function` and the relevant calls to `languaget`